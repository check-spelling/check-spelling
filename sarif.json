{
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
  "version": "2.1.0",
  "properties": {
    "comment": "check-spelling test"
  },
  "runs": [
    {
      "tool": {
        "driver": {
          "name": "check-spelling",
          "informationUri": "https://github.com/check-spelling/check-spelling",
          "rules": [
            {
              "id": "unrecognized-spelling",
              "name": "UnrecognizedSpelling",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#unrecognized-spelling",
              "shortDescription": {
                "text": "Unrecognized Spelling"
              },
              "fullDescription": {
                "text": "Token is neither in the dictionary nor expected"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Correct spelling\n- Add to [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n- Mask using [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n\n"
              },
              "defaultConfiguration": {
                "level": "error"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "error",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "forbidden-pattern",
              "name": "ForbiddenPattern",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#forbidden-pattern",
              "shortDescription": {
                "text": "Forbidden Pattern"
              },
              "fullDescription": {
                "text": "Forbidden pattern matched this text"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Alter text so the pattern does not match\n- Add a [pattern](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns) to patterns.txt that matches (a portion of) the flagged content to accept it\n- Delete, edit, or comment out the pattern\n\n"
              },
              "defaultConfiguration": {
                "level": "error"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "error",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "noisy-file",
              "name": "NoisyFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#noisy-file",
              "shortDescription": {
                "text": "Noisy File"
              },
              "fullDescription": {
                "text": "There were more non-words than dictionary words in the file"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Add more words to the dictionary\n- Use patterns to mask tokens.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "recommendation",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "noisy-file-list",
              "name": "NoisyFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#noisy-file-list",
              "shortDescription": {
                "text": "Noisy File List"
              },
              "fullDescription": {
                "text": "There were more non-words than dictionary words in the file list"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Disable [`check_file_names`](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Add more words from file paths/names to the dictionary\n- Use patterns to mask tokens.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "recommendation",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "large-file",
              "name": "LargeFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#large-file",
              "shortDescription": {
                "text": "Large File"
              },
              "fullDescription": {
                "text": "File is very large"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Adjust 'largest_file' in action configuration if you want to check the file."
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "limited-references",
              "name": "LimitedReferences",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#limited-references",
              "shortDescription": {
                "text": "Limited References"
              },
              "fullDescription": {
                "text": "There were an excessive number of uses of this token. Not all instances are identified."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add word to the [dictionary](https://github.com/check-spelling/check-spelling/wiki/Configuration#allow)\n- Correct the spelling\n- To see more instances, increase action configuration \"unknown_word_limit\".\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "candidate-pattern",
              "name": "CandidatePattern",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#candidate-pattern",
              "shortDescription": {
                "text": "Candidate Pattern"
              },
              "fullDescription": {
                "text": "A line with unrecognized tokens matches a pattern. Patterns enable the tool to ignore content."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the pattern to [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- Remove the pattern from candidates.patterns\n\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "recommendation",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "bad-regex",
              "name": "BadRegex",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#bad-regex",
              "shortDescription": {
                "text": "Bad Regex"
              },
              "fullDescription": {
                "text": "Perl did not like the regular expression pattern."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Correct the [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- See [bad-regex](https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#bad-regex)\n- Remove the pattern.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "error",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "dictionary-not-found",
              "name": "DictionaryNotFound",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#dictionary-not-found",
              "shortDescription": {
                "text": "Dictionary not found"
              },
              "fullDescription": {
                "text": "The specified dictionary could not be found."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Correct the dictionary address.\n- Check the [dictionary_source_prefixes](https://github.com/check-spelling/check-spelling/wiki/Configuration#dictionary_source_prefixes)\n- Remove the entry.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "no-newline-at-eof",
              "name": "NoNewlineAtEOF",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#no-newline-at-eof",
              "shortDescription": {
                "text": "No newline at EOF"
              },
              "fullDescription": {
                "text": "The specified file is missing a newline at the end of the file."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add a blank line to the end of the file.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "recommendation",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "non-alpha-in-dictionary",
              "name": "NonAlphaInDictionary",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#non-alpha-in-dictionary",
              "shortDescription": {
                "text": "Non-alpha in dictionary"
              },
              "fullDescription": {
                "text": "The dictionary entry has non-alpha characters."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Remove non-alpha characters\n- Place each word on its own line\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "recommendation",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "check-file-path",
              "name": "CheckFilePath",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#check-file-path",
              "shortDescription": {
                "text": "Check File Path"
              },
              "fullDescription": {
                "text": "Token in file path is neither in the dictionary nor expected"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\nNote that this check is against the file path not the displayed line.\n- Rename file/directory to correct spelling\n- Add to [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n- Mask using [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n\n"
              },
              "defaultConfiguration": {
                "level": "error"
              },
              "properties": {
                "precision": "high",
                "problem.severity": "error",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "minified-file",
              "name": "MinifiedFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#minified-file",
              "shortDescription": {
                "text": "Minified File"
              },
              "fullDescription": {
                "text": "File has incredibly long lines, perhaps it's minified?"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Add 'minified-file' to 'disable_checks' in the configuration.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "duplicate-extra-dictionary",
              "name": "DuplicateExtraDictionary",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#duplicate-extra-dictionary",
              "shortDescription": {
                "text": "Duplicate Extra Dictionary"
              },
              "fullDescription": {
                "text": "Workflow 'extra_dictionaries' contains multiple entries for the same dictionary"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Remove duplicate entries. check-spelling will ignore duplicate instances.\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "utf16-surrogate",
              "name": "Utf16Surrogate",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#utf16-surrogate",
              "shortDescription": {
                "text": "UTF-16 Surrogate"
              },
              "fullDescription": {
                "text": "File contains a utf-16 surrogate. It's likely not a text file."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "medium",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "binary-file",
              "name": "BinaryFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#binary-file",
              "shortDescription": {
                "text": "Binary File"
              },
              "fullDescription": {
                "text": "File magic determined to be binary. It's likely not a text file."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Disable `use_file_magic`\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "token-is-substring",
              "name": "TokenIsSubstring",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#token-is-substring",
              "shortDescription": {
                "text": "Token is a substring"
              },
              "fullDescription": {
                "text": "While trying to find the token in the line, the exact instance could not be found."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the entry to [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n- Correct any overeager [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- See [Feature: Warning about substring matches](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Warning-about-substring-matches)\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "slow-file",
              "name": "SlowFile",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#slow-file",
              "shortDescription": {
                "text": "Slow file"
              },
              "fullDescription": {
                "text": "Parsing the file exceeded the parsing timeout."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add the entry to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Adjust [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- Adjust the [timeout](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Parsing-Timeouts#configuration)\n- See [Feature: Parsing Timeouts](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Parsing-Timeouts)\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "ignored-expect-variant",
              "name": "IgnoredExpectVariant",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#ignored-expect-variant",
              "shortDescription": {
                "text": "Ignored Expect Variant"
              },
              "fullDescription": {
                "text": "Multiple related word variants listed in expect"
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Remove the entry from [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "no-files-to-check",
              "name": "NoFilesToCheck",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#no-files-to-check",
              "shortDescription": {
                "text": "No files to check"
              },
              "fullDescription": {
                "text": "After applying any excludes and only patterns, no files were found to check."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Remove or fix patterns in [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Remove or broaden patterns in [only](https://github.com/check-spelling/check-spelling/wiki/Configuration#only)\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            },
            {
              "id": "missing-on-pull-request-event",
              "name": "MissingOnPullRequestEvent",
              "helpUri": "https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#missing-on-pull-request-event",
              "shortDescription": {
                "text": "Missing on pull_request event"
              },
              "fullDescription": {
                "text": "Action is configured with `suppress_push_for_open_pull_request` but there is no configured on: pull_request event handler."
              },
              "help": {
                "text": "?",
                "markdown": "**Remediation (click \"Show more\" below)**:\n\n- Add `pull_request` / `pull_request_target` to the `on:` section of the workflow\n- Remove `suppress_push_for_open_pull_request:` from the configuration.\n- If you have the event handler configured and you see this, please file a bug (including your workflow file).\n\n"
              },
              "defaultConfiguration": {
                "level": "warning"
              },
              "properties": {
                "precision": "low",
                "problem.severity": "warning",
                "tags": [
                  "source-code",
                  "code-reviews"
                ]
              }
            }
          ]
        }
      },
      "results": []
    }
  ]
}
