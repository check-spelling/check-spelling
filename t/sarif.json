{
    "runs": [
        {
            "tool": {
                "driver": {
                    "name": "check-spelling",
                    "rules": [
                        {
                            "id": "unrecognized-spelling",
                            "defaultConfiguration": {
                                "level": "warning"
                            },
                            "properties": {
                                "precision": "high",
                                "problem.severity": "warning",
                                "tags": [
                                    "source-code",
                                    "code-reviews"
                                ]
                            }
                        }
                    ]
                }
            }
        }
    ]
}
