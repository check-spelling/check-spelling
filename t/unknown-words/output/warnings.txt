t/unknown-words/input/sample.file:1:23 ... 34, Warning - `sample.file` matches a line_forbidden.patterns entry: `\bsample\.file\b`. (forbidden-pattern)
t/unknown-words/input/sample.file:2:7 ... 11, Warning - `diid` is not a recognized word. (unrecognized-spelling)
t/unknown-words/input/sample.file:2:12 ... 16, Warning - `youu` is not a recognized word. (unrecognized-spelling)
t/unknown-words/input/sample.file:2:17 ... 21, Warning - `fixx` is not a recognized word. (unrecognized-spelling)
t/unknown-words/input/sample.file:2:22 ... 27, Warning - `thiss` is not a recognized word. (unrecognized-spelling)
