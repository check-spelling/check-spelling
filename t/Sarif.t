#!/usr/bin/env -S perl -T -Ilib

use strict;
use warnings;

use File::Basename;
use File::Temp qw/ tempfile /;
use Test::More;

plan tests => 3;
use_ok('CheckSpelling::Sarif');

is(CheckSpelling::Sarif::encode_low_ascii("\x05"), '\u0005');

my $tests = dirname(__FILE__);
my $base = dirname($tests);

$ENV{'CHECK_SPELLING_VERSION'} = '0.0.0';
my ($fh, $sarif_merged, $warnings);
($fh, $warnings) = tempfile();
print $fh 'lib/CheckSpelling/Sarif.pm:3:24 ... 29, Error - `Sarif` is not a recognized word. (unrecognized-spelling)
https://example.com/lib/CheckSpelling/Sarif.pm:3:24 ... 28, Error - `Star` is not a recognized word. (unrecognized-spelling)

';
close $fh;
$ENV{'warning_output'} = $warnings;
($fh, $sarif_merged) = tempfile();
print $fh CheckSpelling::Sarif::main("$base/sarif.json","$tests/sarif.json");
close $fh;
$ENV{'PATH'} = '/bin';
if (-e '/opt/homebrew/bin/go') {
    $ENV{'PATH'} .= ':/opt/homebrew/bin';
}

my $go_bin_output=`go env GOPATH`;
my $jd_output = '';
# if go isn't available, we'll skip this...
if ($go_bin_output) {
    $go_bin_output =~ /(.*)/;
    my $go_bin = $1;
    $ENV{'PATH'} = "/bin:$go_bin/bin";
    my $jd_output = `jd -set '$sarif_merged' '$tests/sarif.json.expected'`;
}
is($jd_output, '');
