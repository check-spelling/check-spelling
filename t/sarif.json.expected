{
"runs":[
{
"results":[
{
"locations":[
{
"physicalLocation":{
"artifactLocation":{
"uri":"lib/CheckSpelling/Sarif.pm",
"uriBaseId":"%SRCROOT%"
},
"region":{
"endColumn":29,
"startColumn":24,
"startLine":3
}
}
}
],
"message":{
"text":"[Sarif](#security-tab) is not a recognized word. \\(unrecognized-spelling\\)"
},
"ruleId":"unrecognized-spelling",
"ruleIndex":0
}
],
"tool":{
"driver":{
"version":"0.0.0",
"name":"check-spelling",
"informationUri":"https://github.com/check-spelling/check-spelling",
"rules":[
{
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"problem.severity":"warning",
"precision":"high"
},
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Correct spelling\n- Add to [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n- Mask using [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n\n"
},
"fullDescription":{
"text":"Token is neither in the dictionary nor expected"
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#unrecognized-spelling",
"shortDescription":{
"text":"Unrecognized Spelling"
},
"name":"UnrecognizedSpelling",
"id":"unrecognized-spelling"
},
{
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Alter text so the pattern does not match\n- Add a [pattern](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns) to patterns.txt that matches (a portion of) the flagged content to accept it\n- Delete, edit, or comment out the pattern\n\n"
},
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"precision":"high",
"problem.severity":"error"
},
"fullDescription":{
"text":"Forbidden pattern matched this text"
},
"defaultConfiguration":{
"level":"error"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#forbidden-pattern",
"shortDescription":{
"text":"Forbidden Pattern"
},
"id":"forbidden-pattern",
"name":"ForbiddenPattern"
},
{
"shortDescription":{
"text":"Noisy File"
},
"id":"noisy-file",
"name":"NoisyFile",
"help":{
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Add more words to the dictionary\n- Use patterns to mask tokens.\n\n",
"text":"?"
},
"properties":{
"problem.severity":"recommendation",
"precision":"high",
"tags":[
"code-reviews",
"source-code"
]
},
"fullDescription":{
"text":"There were more non-words than dictionary words in the file"
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#noisy-file"
},
{
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#large-file",
"help":{
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Adjust 'largest_file' in action configuration if you want to check the file.",
"text":"?"
},
"properties":{
"problem.severity":"warning",
"precision":"medium",
"tags":[
"source-code",
"code-reviews"
]
},
"fullDescription":{
"text":"File is very large"
},
"name":"LargeFile",
"id":"large-file",
"shortDescription":{
"text":"Large File"
}
},
{
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#limited-references",
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add word to the [dictionary](https://github.com/check-spelling/check-spelling/wiki/Configuration#allow)\n- Correct the spelling\n- To see more instances, increase action configuration \"unknown_word_limit\".\n\n"
},
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"precision":"high",
"problem.severity":"warning"
},
"fullDescription":{
"text":"There were an excessive number of uses of this token. Not all instances are identified."
},
"name":"LimitedReferences",
"id":"limited-references",
"shortDescription":{
"text":"Limited References"
}
},
{
"shortDescription":{
"text":"Candidate Pattern"
},
"name":"CandidatePattern",
"id":"candidate-pattern",
"properties":{
"problem.severity":"recommendation",
"precision":"medium",
"tags":[
"code-reviews",
"source-code"
]
},
"help":{
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the pattern to [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- Remove the pattern from candidates.patterns\n\n\n",
"text":"?"
},
"fullDescription":{
"text":"A line with unrecognized tokens matches a pattern. Patterns enable the tool to ignore content."
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#candidate-pattern"
},
{
"shortDescription":{
"text":"Bad Regex"
},
"name":"BadRegex",
"id":"bad-regex",
"help":{
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Correct the [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n- See [bad-regex](https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#bad-regex)\n- Remove the pattern.\n\n",
"text":"?"
},
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"precision":"medium",
"problem.severity":"error"
},
"fullDescription":{
"text":"Perl did not like the regular expression pattern."
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#bad-regex"
},
{
"help":{
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Correct the dictionary address.\n- Check the [dictionary_source_prefixes](https://github.com/check-spelling/check-spelling/wiki/Configuration#dictionary_source_prefixes)\n- Remove the entry.\n\n",
"text":"?"
},
"properties":{
"precision":"high",
"problem.severity":"warning",
"tags":[
"code-reviews",
"source-code"
]
},
"fullDescription":{
"text":"The specified dictionary could not be found."
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#dictionary-not-found",
"shortDescription":{
"text":"Dictionary not found"
},
"id":"dictionary-not-found",
"name":"DictionaryNotFound"
},
{
"shortDescription":{
"text":"No newline at EOF"
},
"id":"no-newline-at-eof",
"name":"NoNewlineAtEOF",
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add a blank line to the end of the file.\n\n"
},
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"precision":"high",
"problem.severity":"recommendation"
},
"fullDescription":{
"text":"The specified file is missing a newline at the end of the file."
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#no-newline-at-eof"
},
{
"shortDescription":{
"text":"Check File Path"
},
"name":"CheckFilePath",
"id":"check-file-path",
"fullDescription":{
"text":"Token in file path is neither in the dictionary nor expected"
},
"properties":{
"tags":[
"source-code",
"code-reviews"
],
"precision":"high",
"problem.severity":"error"
},
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\nNote that this check is against the file path not the displayed line.\n- Rename file/directory to correct spelling\n- Add to [expect](https://github.com/check-spelling/check-spelling/wiki/Configuration#expect)\n- Mask using [patterns](https://github.com/check-spelling/check-spelling/wiki/Configuration#patterns)\n\n"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#check-file-path",
"defaultConfiguration":{
"level":"error"
}
},
{
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#minified-file",
"defaultConfiguration":{
"level":"warning"
},
"fullDescription":{
"text":"File has incredibly long lines, perhaps it's minified?"
},
"properties":{
"problem.severity":"warning",
"precision":"medium",
"tags":[
"code-reviews",
"source-code"
]
},
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Add 'minified-file' to 'disable_checks' in the configuration.\n\n"
},
"name":"MinifiedFile",
"id":"minified-file",
"shortDescription":{
"text":"Minified File"
}
},
{
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#duplicate-extra-dictionary",
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Remove duplicate entries. check-spelling will ignore duplicate instances.\n\n"
},
"properties":{
"precision":"medium",
"problem.severity":"warning",
"tags":[
"code-reviews",
"source-code"
]
},
"fullDescription":{
"text":"Workflow 'extra_dictionaries' contains multiple entries for the same dictionary"
},
"id":"duplicate-extra-dictionary",
"name":"DuplicateExtraDictionary",
"shortDescription":{
"text":"Duplicate Extra Dictionary"
}
},
{
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#utf16-surrogate",
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n\n"
},
"properties":{
"tags":[
"code-reviews",
"source-code"
],
"precision":"medium",
"problem.severity":"warning"
},
"fullDescription":{
"text":"File contains a utf-16 surrogate. It's likely not a text file."
},
"id":"utf16-surrogate",
"name":"Utf16Surrogate",
"shortDescription":{
"text":"UTF-16 Surrogate"
}
},
{
"shortDescription":{
"text":"Binary File"
},
"id":"binary-file",
"name":"BinaryFile",
"help":{
"text":"?",
"markdown":"**Remediation (click \"Show more\" below)**:\n\n- Add the file to [excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration#excludes)\n- Disable `use_file_magic`\n\n"
},
"properties":{
"tags":[
"source-code",
"code-reviews"
],
"precision":"low",
"problem.severity":"warning"
},
"fullDescription":{
"text":"File magic determined to be binary. It's likely not a text file."
},
"defaultConfiguration":{
"level":"warning"
},
"helpUri":"https://github.com/check-spelling/check-spelling/wiki/Event-descriptions#binary-file"
}
]
}
}
}
],
"properties":{
"comment":"check-spelling test"
},
"$schema":"https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
"version":"2.1.0"
}
